library/core/src/iter/mod.rs

```diff
@@ -368,6 +368,7 @@ macro_rules! impl_fold_via_try_fold {
         impl_fold_via_try_fold! { @internal spec_rfold -> spec_try_rfold }
     };
     (@internal $fold:ident -> $try_fold:ident) => {
+        #[allow(non_camel_case_types)]
         #[inline]
         fn $fold<AAA, FFF>(mut self, init: AAA, fold: FFF) -> AAA
```

Comment:
```suggestion
        fn $fold<A, F>(mut self, init: A, fold: F) -> A
```
Not sure why the weird names are used here.

Reply:
Seems like it's because `A` and `F` are already used as generic parameters elsewhere:
```
error[E0403]: the name `A` is already used for a generic parameter in this item's generic parameters
    --> library\core\src\iter\mod.rs:375:18
     |
357  | / macro_rules! impl_fold_via_try_fold {
358  | |     (fold -> try_fold) => {
359  | |         impl_fold_via_try_fold! { @internal fold -> try_fold }
     | |         ------------------------------------------------------ in this macro invocation (#2)
360  | |     };
...    |
375  | |         fn $fold<A, F>(mut self, init: A, fold: F) -> A
     | |                  ^ already used
...    |
383  | |     };
384  | | }
     | | -
     | | |
     | |_in this expansion of `impl_fold_via_try_fold!` (#1)
     |   in this expansion of `impl_fold_via_try_fold!` (#2)
     |
    ::: library\core\src\iter\range.rs:1219:6
     |
1219 |   impl<A: Step> Iterator for ops::RangeInclusive<A> {
     |        - first use of `A`
...
1288 |       impl_fold_via_try_fold! { fold -> try_fold }
     |       -------------------------------------------- in this macro invocation (#1)
```

For now I've kept them as `allow`ed and added a comment explaining that `A` and `F` are already used as generic parameters in impls where this macro appears.